ugvc/mrd/ppmSeq_utils.py

```diff
@@ -1071,26 +1074,40 @@ def collect_statistics(
     )
     if is_v7_dumbell:
         df_dumbell_leftover = read_dumbell_leftover_from_trimmer_histogram(
-            trimmer_histogram_extra_csv, legacy_histogram_column_names=legacy_histogram_column_names
+            trimmer_histogram_extra_csv,
+            legacy_histogram_column_names=legacy_histogram_column_names,
         )
         df_stats_shortlist = pd.concat((df_stats_shortlist, df_dumbell_leftover))
 
     # save
     if not output_filename.endswith(".h5"):
         output_filename += ".h5"
     with pd.HDFStore(output_filename, "w") as store:
+        keys_to_convert = [
+            "stats_shortlist",
+            "sorter_stats",
+            "strand_ratio_category_counts",
+            "strand_ratio_category_norm",
+        ]
         store["stats_shortlist"] = df_stats_shortlist
         store["sorter_stats"] = df_sorter_stats
         store["trimmer_histogram"] = df_trimmer_histogram
         store["strand_ratio_category_counts"] = df_strand_ratio_category
         store["strand_ratio_category_norm"] = df_strand_ratio_category / df_strand_ratio_category.sum()
         if adapter_in_both_ends:
-            store["df_category_concordance"] = df_category_concordance
-            store["df_category_consensus"] = df_category_consensus
+            store["strand_ratio_category_concordance"] = df_category_concordance
+            store["strand_ratio_category_consensus"] = df_category_consensus
+            keys_to_convert += [
+                "strand_ratio_category_concordance",
+                "strand_ratio_category_consensus",
+            ]
         if trimmer_failure_codes_csv:
-            store["df_trimmer_failure_codes"] = df_trimmer_failure_codes
+            store["trimmer_failure_codes"] = df_trimmer_failure_codes
+            keys_to_convert += ["trimmer_failure_codes"]
```

Comment:
I wonder if you need trimmer_failure_codes in your h5... When running on-tool, these should be uploaded anyway. Same for sorter stats (unless you are making changes)